src/lib/core/assoc.h

```diff
@@ -189,6 +189,76 @@ mh_strnu32_find_str(struct mh_strnu32_t *h, const char *str, uint32_t len)
 	return mh_strnu32_find(h, &key, NULL);
 }
 
+/*
+ * Map: (const char *, uint32_t, const char *, uint32_t) => (void *)
+ */
+
+static inline uint32_t
+mh_access_hash(const char *uname, uint32_t uname_len,
+	       const char *fname, uint32_t fname_len)
+{
+	return mh_strn_hash(uname, uname_len) ^ mh_strn_hash(fname, fname_len);
+}
+
+#define mh_name _access
```

Comment:
In my understanding the `assoc.h` file provides general-purpose containers. Taking it into account, I would name the hash and the related functions/structures with something derived from the 'two strings with length' idea, without any relation to the accesses.

Something like the following (I use `strn_strn` name here):

```diff
diff --git a/src/box/rt_lua_call_access.c b/src/box/rt_lua_call_access.c
index 1bf4904713..d0f43c4bfc 100644
--- a/src/box/rt_lua_call_access.c
+++ b/src/box/rt_lua_call_access.c
@@ -4,7 +4,7 @@
 #include "session.h"
 #include "../lua/init.h" /* tarantool_lua_is_builtin_global */
 
-static struct mh_access_t *user_rt_access_cache = NULL;
+static struct mh_strn_strn_t *user_rt_access_cache = NULL;
 static struct mh_strnptr_t *user_rt_universe_access_cache = NULL;
 
 const char *
@@ -19,14 +19,14 @@ user_name_by_id(uint32_t uid, uint32_t *name_len)
 void
 rt_lua_call_access_init(void)
 {
-	user_rt_access_cache = mh_access_new();
+	user_rt_access_cache = mh_strn_strn_new();
 	user_rt_universe_access_cache = mh_strnptr_new();
 }
 
 void
 rt_lua_call_access_destroy(void)
 {
-	mh_access_delete(user_rt_access_cache);
+	mh_strn_strn_delete(user_rt_access_cache);
 	mh_strnptr_delete(user_rt_universe_access_cache);
 }
 
@@ -39,19 +39,18 @@ grant_rt_access(const char *uname, uint32_t uname_len,
 		uint32_t hash = mh_strn_hash(uname, uname_len);
 		const struct mh_strnptr_node_t uid_node =
 			{ uname, uname_len, hash, NULL };
-		struct mh_strnptr_node_t repl_uid_node = { NULL, 0, 0, NULL };
-		struct mh_strnptr_node_t *prepl_uid_node = &repl_uid_node;
+		struct mh_strnptr_node_t *ignored;
 		mh_strnptr_put(user_rt_universe_access_cache, &uid_node,
-			       &prepl_uid_node, NULL);
+			       &ignored, NULL);
 		return;
 	}
 
 	/* Grant access to lua_call function. */
-	uint32_t hash = mh_access_hash(uname, uname_len, fname, fname_len);
-	const struct mh_access_node_t access_node =
+	uint32_t hash = mh_strn_strn_hash(uname, uname_len, fname, fname_len);
+	const struct mh_strn_strn_node_t access_node =
 		{ uname, uname_len, fname, fname_len, hash };
-	struct mh_access_node_t *ignored;
-	mh_access_put(user_rt_access_cache, &access_node, &ignored, NULL);
+	struct mh_strn_strn_node_t *ignored;
+	mh_strn_strn_put(user_rt_access_cache, &access_node, &ignored, NULL);
 }
 
 void
@@ -67,13 +66,13 @@ revoke_rt_access(const char *uname, uint32_t uname_len,
 				       NULL);
 	} else {
 		/* Revoke access to lua_call function. */
-		uint32_t hash = mh_access_hash(uname, uname_len, fname,
+		uint32_t hash = mh_strn_strn_hash(uname, uname_len, fname,
 					       fname_len);
-		struct mh_access_node_t key =
+		struct mh_strn_strn_node_t key =
 			{ uname, uname_len, fname, fname_len, hash };
-		mh_int_t pos = mh_access_find(user_rt_access_cache, &key, NULL);
+		mh_int_t pos = mh_strn_strn_find(user_rt_access_cache, &key, NULL);
 		if (pos != mh_end(user_rt_access_cache))
-			mh_access_del(user_rt_access_cache, pos, NULL);
+			mh_strn_strn_del(user_rt_access_cache, pos, NULL);
 	}
 }
 
@@ -88,10 +87,10 @@ check_rt_access(const char *uname, uint32_t uname_len,
 	    !tarantool_lua_is_builtin_global(fname, fname_len))
 		return RT_ACCESS;
 
-	uint32_t hash = mh_access_hash(uname, uname_len, fname, fname_len);
-	struct mh_access_node_t key =
+	uint32_t hash = mh_strn_strn_hash(uname, uname_len, fname, fname_len);
+	struct mh_strn_strn_node_t key =
 		{ uname, uname_len, fname, fname_len, hash };
-	pos = mh_access_find(user_rt_access_cache, &key, NULL);
+	pos = mh_strn_strn_find(user_rt_access_cache, &key, NULL);
 	if (pos == mh_end(user_rt_access_cache))
 		return RT_NO_ACCESS;
 	return RT_ACCESS;
diff --git a/src/lib/core/assoc.h b/src/lib/core/assoc.h
index 5f63a47776..a30f2c36d5 100644
--- a/src/lib/core/assoc.h
+++ b/src/lib/core/assoc.h
@@ -194,50 +194,50 @@ mh_strnu32_find_str(struct mh_strnu32_t *h, const char *str, uint32_t len)
  */
 
 static inline uint32_t
-mh_access_hash(const char *uname, uint32_t uname_len,
-	       const char *fname, uint32_t fname_len)
+mh_strn_strn_hash(const char *s1, uint32_t s1_len,
+		  const char *s2, uint32_t s2_len)
 {
-	return mh_strn_hash(uname, uname_len) ^ mh_strn_hash(fname, fname_len);
+	return mh_strn_hash(s1, s1_len) ^ mh_strn_hash(s2, s2_len);
 }
 
-#define mh_name _access
+#define mh_name _strn_strn
 
 /**
- * Node of `mh_access_node_t` hash table.
+ * Node of `mh_strn_strn_node_t` hash table.
  */
-struct mh_access_node_t {
-	/* Key user name. */
-	const char *uname;
-	/* Key user name length. */
-	uint32_t uname_len;
-	/* Key function name. */
-	const char *fname;
+struct mh_strn_strn_node_t {
+	/* First string. */
+	const char *s1;
+	/* First string length. */
+	uint32_t s1_len;
+	/* Second string. */
+	const char *s2;
 	/* Key function name length. */
-	uint32_t fname_len;
-	/* Key hash calculated using `mh_access_hash`. */
+	uint32_t s2_len;
+	/* Key hash calculated using `mh_strn_strn_hash`. */
 	uint32_t hash;
 };
 
-#define mh_key_t struct mh_access_node_t *
-#define mh_node_t struct mh_access_node_t
+#define mh_key_t struct mh_strn_strn_node_t *
+#define mh_node_t struct mh_strn_strn_node_t
 
 #define mh_arg_t void *
 #define mh_hash(a, arg) ((a)->hash)
 #define mh_hash_key(a, arg) ((a)->hash)
-#define mh_cmp(a, b, arg) ((a)->uname_len != (b)->uname_len || \
-			    (a)->fname_len != (b)->fname_len || \
-			    strncmp((a)->uname, (b)->uname, (a)->uname_len) || \
-			    strncmp((a)->fname, (b)->fname, (a)->fname_len))
+#define mh_cmp(a, b, arg) ((a)->s1_len != (b)->s1_len || \
+			    (a)->s2_len != (b)->s2_len || \
+			    memcmp((a)->s1, (b)->s1, (a)->s1_len) || \
+			    memcmp((a)->s2, (b)->s2, (a)->s2_len))
 #define mh_cmp_key(a, b, arg) mh_cmp(a, b, arg)
 #include "salad/mhash.h"
 
 static inline mh_int_t
-mh_access_find_str(struct mh_access_t *h, const char *uname, uint32_t uname_len,
-		   const char *fname, uint32_t fname_len)
+mh_strn_strn_find_str(struct mh_strn_strn_t *h, const char *s1, uint32_t s1_len,
+		      const char *s2, uint32_t s2_len)
 {
-	uint32_t hash = mh_access_hash(uname, uname_len, fname, fname_len);
-	struct mh_access_node_t key = {uname, uname_len, fname, fname_len, hash};
-	return mh_access_find(h, &key, NULL);
+	uint32_t hash = mh_strn_strn_hash(s1, s1_len, s2, s2_len);
+	struct mh_strn_strn_node_t key = {s1, s1_len, s2, s2_len, hash};
+	return mh_strn_strn_find(h, &key, NULL);
 }
 
 #if defined(__cplusplus)
```

Reply:
Renamed to `*_strn_strn_*`